packages/react/src/helpers/utils.ts

```diff
@@ -1,8 +1,30 @@
+import isEmpty from 'lodash/isEmpty';
+import isArray from 'lodash/isArray';
+import isObject from 'lodash/isObject';
+
 export const isDevelopment = () => process.env.NODE_ENV !== 'production';
 
+const isEmptyObj = (val: any) => isObject(val) && isEmpty(val);
+
+const isEmptyArr = (val: any) => isArray(val) && isEmpty(val);
+
 export const areArraysEqual = (arr1: Array<any>, arr2: Array<any>) => {
```

Comment:
We **may** want to rename this util since two arrays that are empty are not equal due to identity

Reply:
Changing it to `AreArrayValuesEqual` as per offline discussion. Technically we don't do deep equality, but this conveys the point that we're interested in the values!

Reply:
Done here! [ac5153d](https://github.com/aws-amplify/amplify-ui/pull/1619/commits/ac5153db3fb572ffa761f07f70280a0ca056e896)